### PR TITLE
ci: Add issue generation to ZD Test pipeline

### LINKED ISSUE #8212: Core :: Automate issue generation for scheduled ZD run
Now that we officially released a ZD version, it would be good to have issues as soon as our changes break the zero downtime tests.

## DoD

- [ ] Similarly to integration test the zd pipeline generates an issue in the case of redrun
- [ ] The API probe failures are ignored if under an acceptable limit 
- [ ] Scheduled run is green

### PR DESCRIPTION
**Pull request by RealAnna**
## This PR

- adds automated issue generation to ZD test

### Related Issues

Fixes #8212 

### How to test

image

image

eg. https://github.com/keptn/keptn/runs/7976459016?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/zero-downtime-tests.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: ZeroDowntime Tests
 on:
   schedule:
     # * is a special character in YAML so you have to quote this string
-    - cron: '0 1 * * 1' # run zero downtime tests at 1 AM (UTC), every monday (1)
+    - cron:  '0 4 * * 1-5' # run integration tests at 4 AM (UTC), monday to friday (1-5)
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:5]: was this change intentional?

File: .github/workflows/zero-downtime-tests.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ on:
       upgradeTo:
         description: 'The helm upgrade command will run with this version (e.g. 0.15.0-dev-PR-7504 or a full address like in the default value )'
         required: true
-        default: 'https://github.com/keptn/helm-charts-dev/raw/gh-pages/packages/keptn-0.18.0-dev.tgz'
+        default: 'https://github.com/keptn/helm-charts-dev/raw/gh-pages/packages/keptn-0.18.1-dev.tgz'
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:15]: you could also use:
```suggestion
        default: 'https://charts.keptn.sh/packages/keptn-0.18.1-dev.tgz'
```

File: .github/workflows/zero-downtime-tests.yml [hunk 3]
```diff
@@ -57,6 +57,11 @@ jobs:
       KUBECONFIG: ${{ matrix.KUBECONFIG }}
       KEPTN_NAMESPACE: "keptn-zd-test-${{ github.run_number }}-${{ github.run_attempt }}"
       GO_VERSION: "~1.18"
+      RUN_CONTINUOUS_DELIVERY_TEST: "true"
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:60]: is this variable still necessary, since we removed the CD test already?

File: .github/workflows/zero-downtime-tests.yml [hunk 4]
```diff
@@ -313,50 +306,98 @@ jobs:
...
+              npm run test:ui >> /tmp/uitest.log 2>&1
             done
           else
             echo "UI tests will not run"
           fi
         
           wait $GOTEST
           if [[ $? -ne 0 ]]; then
-            echo "::error::Zero Downtime go test failed - exiting"
+            echo " Zero Downtime go test failed - exiting "
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:341]: [nitpick]
```suggestion
            echo " Zero Downtime go test failed - exiting"
```

File: .github/workflows/zero-downtime-tests.yml [hunk 5]
```diff
@@ -313,50 +306,98 @@ jobs:
...
+      - name: Generate XML
+        if: always()
+        run: go-junit-report -in temp_res/gotest.log -parser gojson -out temp_res/gotest.xml
+
+      - name: Generate test Summary
+        uses: test-summary/action@v1
+        with:
+          paths: "temp_res/*.xml"
+          output: temp_res/test-summary.md
+        if: always()
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:356]: We always have the `if` at the top :)
```suggestion
      - name: Generate Test Summary
        if: always()
        uses: test-summary/action@v1
        with:
          paths: "temp_res/*.xml"
          output: temp_res/test-summary.md
```

File: .github/workflows/zero-downtime-tests.yml [hunk 6]
```diff
@@ -313,50 +306,98 @@ jobs:
...
+          FINAL=$(cat temp_res/test-summary.md) 
+          echo "$FINAL" >>  $GITHUB_STEP_SUMMARY
+          echo "##[set-output name=FINAL;]$(echo ${FINAL})"
+
+      - name: Generate Github Summary
+        if: always()
+        env:
+          TESTUI: ${{ github.event.inputs.testUI }}
+          UIOUT: ${{ steps.log_output.outcome }}
+        run: |
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:383]:
```suggestion
        run: |
```

File: .github/workflows/zero-downtime-tests.yml [hunk 7]
```diff
@@ -313,50 +306,98 @@ jobs:
...
+          echo "##[set-output name=FINAL;]$(echo ${FINAL})"
+
+      - name: Generate Github Summary
+        if: always()
+        env:
+          TESTUI: ${{ github.event.inputs.testUI }}
+          UIOUT: ${{ steps.log_output.outcome }}
+        run: |  
+          if [[ $TESTUI == 'true' && $UIOUT != 'success' ]]; then
+             echo "UI test failed!" >> $GITHUB_STEP_SUMMARY
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:385]:
```suggestion
             echo "UI test failed!" >> $GITHUB_STEP_SUMMARY
```

File: .github/workflows/zero-downtime-tests.yml [hunk 8]
```diff
@@ -313,50 +306,98 @@ jobs:
...
+             echo "* Triggered by: @$GITHUB_ACTOR" >> $GITHUB_STEP_SUMMARY
+          fi
+          
+          
+          echo "* Branch: $BRANCH" >> $GITHUB_STEP_SUMMARY
+          echo "* Version: $VERSION" >> $GITHUB_STEP_SUMMARY
+          echo "* Datetime: $DATETIME" >> $GITHUB_STEP_SUMMARY
+          echo "* Commit: $GIT_SHA" >> $GITHUB_STEP_SUMMARY
+          echo "* Upgraded to: $UPGRADE_HELM_CHART" >> $GITHUB_STEP_SUMMARY
+          echo "" >>$GITHUB_STEP_SUMMARY
```
Comment (mowies) [inline .github/workflows/zero-downtime-tests.yml:399]:
```suggestion
          echo "" >> $GITHUB_STEP_SUMMARY
```

File: .github/workflows/zero-downtime-tests.yml (hunk 1 above)
Comment (RealAnna) [reply .github/workflows/zero-downtime-tests.yml:5]: yes we discussed to have the test daily, but I'll rename it to zd

File: .github/workflows/zero-downtime-tests.yml (hunk 3 above)
Comment (RealAnna) [reply .github/workflows/zero-downtime-tests.yml:60]: I'll remove it

**Review by mowies — APPROVED** [review]
LGTM!
